src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,88 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
+    private final Map<Class<?>, Class<?>> implementationsMap = new HashMap<>();
+
+    private Injector() {
+        implementationsMap.put(ProductService.class, ProductServiceImpl.class);
+        implementationsMap.put(ProductParser.class, ProductParserImpl.class);
+        implementationsMap.put(FileReaderService.class, FileReaderServiceImpl.class);
+    }
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
+        Class<?> clazz = findImplementation(interfaceClazz);
+        if (clazz == null) {
+            throw new RuntimeException("No implementation found for class: "
+                    + interfaceClazz.getName());
+        }
+
+        return createInstance(clazz);
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        if (interfaceClazz != null && interfaceClazz.isInterface()) {
+            return implementationsMap.get(interfaceClazz);
+        }
         return null;
```

Comment:
If `interfaceClazz` is not an interface, you might want to handle this case explicitly, perhaps by returning the class itself or throwing an exception, depending on your design requirements.